lib/msprime.c

```diff
@@ -3169,15 +3140,14 @@ msp_merge_ancestors(msp_t *self, avl_tree_t *Q, population_id_t population_id,
                 }
             }
         } else {
-            if (!coalescence) {
+            if (new_node_id==TSK_NULL) {
                 coalescence = true;
                 l_min = l;
-                ret = msp_store_node(self, 0, self->time, population_id, individual);
+                new_node_id = (tsk_id_t) msp_store_node(self, 0, self->time, population_id, individual);
```

Comment:
Need to check the error here:
```c
if (new_node_id < 0) {
    ret = (int) new_node_id;
    goto out;
}
```

Reply:
I think you also need to modify msp_store_node to return the node ID.

Reply:
I think tsk_node_table_add_row already returns the node ID to msp_store_node (see the return of tsk_node_table_add_row_internal)